packages/opencensus-instrumentation-redis/test/test-redis.ts

```diff
@@ -0,0 +1,217 @@
+ * instances from.
+ * @param expectedName The expected name of the first root span.
+ * @param expectedKind The expected kind of the first root span.
+ */
+function assertSpan(
+    rootSpanVerifier: RootSpanVerifier, expectedName: string,
+    expectedKind: SpanKind, verifyAttribute?: (span: Span) => boolean) {
+  assert.strictEqual(rootSpanVerifier.endedRootSpans.length, 1);
+  assert.strictEqual(rootSpanVerifier.endedRootSpans[0].spans.length, 1);
+  assert.strictEqual(
+      rootSpanVerifier.endedRootSpans[0].spans[0].name, expectedName);
+  assert.strictEqual(
+      rootSpanVerifier.endedRootSpans[0].spans[0].kind, expectedKind);
+  if (typeof verifyAttribute === 'function') {
+    for (const span of rootSpanVerifier.endedRootSpans[0].spans) {
+      assert(verifyAttribute(span), 'failed to verify attribute');
+    }
+  }
+}
+
+describe('RedisPlugin', () => {
+  // For these tests, mongo must be runing. Add OPENCENSUS_REDIS_TESTS to run
+  // these tests.
+  const OPENCENSUS_REDIS_TESTS = process.env.OPENCENSUS_REDIS_TESTS as string;
+  const OPENCENSUS_REDIS_HOST = process.env.OPENCENSUS_REDIS_HOST as string;
+  let shouldTest = true;
+  if (!OPENCENSUS_REDIS_TESTS) {
+    console.log('Skipping test-redis. Run REDIS to test');
+    shouldTest = false;
+  }
+
+  const URL = `redis://${OPENCENSUS_REDIS_HOST || 'localhost'}:6379`;
+  const VERSION = '2.8.0';
+  const REDIS_QUERY_TYPE = SpanKind.CLIENT;
+
+  const tracer = new CoreTracer();
+  const rootSpanVerifier = new RootSpanVerifier();
+  let client: redis.RedisClient;
+
+  before((done) => {
+    tracer.start({samplingRate: 1, logger: logger.logger(1)});
+    tracer.registerSpanEventListener(rootSpanVerifier);
+    plugin.enable(redis, tracer, VERSION, {}, '');
+    client = redis.createClient({url: URL});
+    client.on('error', (err: Error) => {
+      console.log(
+          'Skipping test-redis. Could not connect. Run Redis to test', err);
+      shouldTest = false;
+      done();
+    });
+    client.on('ready', done);
+  });
+
+  beforeEach(function redisBeforeEach(done) {
+    // Skiping all tests in beforeEach() is a workarround. Mocha does not work
+    // properly when skiping tests in before() on nested describe() calls.
+    // https://github.com/mochajs/mocha/issues/2819
+    if (!shouldTest) {
+      this.skip();
+    }
+    rootSpanVerifier.endedRootSpans = [];
+    // Non traced insertion of basic data to perform tests
+    client.set('test', 'data');
+    return done();
+  });
+
+  afterEach((done) => {
+    client.del('hash', done);
+  });
+
+  after(() => {
+    if (client) {
+      client.quit();
+    }
+  });
+
+  /** Should intercept query */
+  describe('Instrumenting query operations', () => {
+    it('should create a child span for hset', (done) => {
+      tracer.startRootSpan({name: 'insertRootSpan'}, (rootSpan: RootSpan) => {
+        // tslint:disable-next-line:no-any
+        client.hset('hash', 'random', 'random', (err: Error, result: any) => {
```

Comment:
It doesn't look like `result` is used below. Can you use the empty type `{}` and remove the tslint disable above?